### ISSUE TITLE
IDynamicDebugTargetsGenerator - vs2015

### ISSUE THREAD

**Issue by dazinator** [Microsoft/VSProjectSystem#209]
I have been able to look at the roslyn project system to work out how to implement an `IDynamicDebugTargetsGenerator` that adds targets to the start button drowndown.

I managed to get this working great for vs2017.

However when I tried to do the same thing for a VS2015 cps extension, I can't get it to work. Basically my `IDynamicDebugTargetsGenerator` is never invoked to provide the additional targets.

I have been through everything I can think of, and originally logged an issue here with all the detail: https://github.com/Microsoft/extendvs/issues/43

My best guess is that the `vsct` that I grabbed from Roslyn project system, works for vs2017, but perhaps needs to be written differently for vs2015 - and I can't find a vs2015 implementation anywhere to use as a reference for this..

Is it possible we could document this extension point, for vs2015 and vs2017 including:

1. Implementing and exporting `IDynamicDebugTargetsGenerator`
2. Imlementing the corresponding `vsct` so that commands are correctly routed.

Some working samples would be even better.

A basic working sample for each of the main CPS extensions points would be a great reference too.

I'd be happy to help contribute to that.

**Comment by adrianvmsft** [comment]
It looks like roslyn project system has an [implementation](https://github.com/dotnet/roslyn-project-system/blob/master/src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/DebugProfileDebugTargetGenerator.cs).
Here is the corresponding [vsct](https://github.com/dotnet/roslyn-project-system/blob/master/src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/Menus.vsct)

**Comment by dazinator** [comment]
Yeah I'm not confident this works for vs2015 though - well i haven't been able to get it to work for vs2025. The vsct seems to be a bit magical imho. I am left wondering whether the vsct needs to be different for vs2015.

I'll can put together a sample for vs2017 and vs2015 to show you what i mean?
